### PR TITLE
add teams chats selectors boilerplate

### PR DESCRIPTION
**Pull request by ryanfkeepers**
standard boilerplate additions, largely copy-paste and renamed from other files.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :sunflower: Feature

#### Issue(s)

* #5061

#### Test Plan

- [x] :zap: Unit test

### CODE REVIEW

**Review by pandeyabs — APPROVED** [review]

File: src/pkg/backup/details/teamsChats.go
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/pkg/backup/details/teamsChats.go:1]: punting: I'll come back to this at the end of all the PRs to minimize churn.

File: src/pkg/selectors/teamsChats.go [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+	"github.com/alcionai/corso/src/pkg/backup/identity"
+	"github.com/alcionai/corso/src/pkg/fault"
+	"github.com/alcionai/corso/src/pkg/filters"
+	"github.com/alcionai/corso/src/pkg/path"
+)
+
+// ---------------------------------------------------------------------------
+// Selectors
+// ---------------------------------------------------------------------------
+
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/pkg/selectors/teamsChats.go:20]: easier said than done, due to the interface design.  Also a very low priority.  Feel free to make a tech debt ticket for it, if wanted.  No harm in tracking the idea.

File: src/pkg/backup/details/teamsChats.go
[the remark this answers is not in the document]
Comment (pandeyabs) [reply src/pkg/backup/details/teamsChats.go:1]: sounds good

File: src/pkg/selectors/teamsChats.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (pandeyabs) [reply src/pkg/selectors/teamsChats.go:20]: https://github.com/alcionai/corso/issues/5129

**Pull request merged**